apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -144,7 +162,16 @@ export class CredentialsService implements CredentialVerifier {
         JSON.stringify(provePresentationDto.presentation),
         JSON.stringify(proofOptions),
         JSON.stringify(key)
-      )
+      ).catch((err) => {
+        if (typeof err === 'string') {
+          //TODO: discuss if BadRequestException should be thrown here instead?
+          throw new InternalServerErrorException(
+            `@spruceid/didkit-wasm-node.issuePresentation error: ${err}`
```

Comment:
We'll need to check how many lines of the repeated code will be reduced after this. If only one or two lines I would keep it as is because it is much easier to understand than the idea behind the `didKitExecutor`. I'll try this for sure.

Reply:
To me, it's not so much about reducing the lines of code and more about encapsulating the duplicated code and comments around the code.

I'm curious, what about the function makes it challenging to understand? Maybe we can make it easier to understand 😄 . I agree it's not a super simple approach. Is it mainly the use of generics that someone reading might not be familiar with?
However, to me, communicating the "why" idea behind `didKitExecutor` can be solved by adding method documentation.

Reply:
I am just not sure if for someone who reads this code this will be another "oh, what is going on here?" thing until he goes to this function definition and sees the comment, I'll try to reduce repeated code and keep it also easy to understand this wrapper. Maybe in a separate PR to discuss this. I am adding TODOs to not forget.

I agree that logic should be defined in one place.

Reply:
> Maybe in a separate PR to discuss this.

Definitely, if you think this would be better, we can do it this way. Up to you. 👍